Assets/Scripts/SS3D/Systems/IngameConsoleSystem/CommandsController.cs

```diff
@@ -29,33 +40,83 @@ public CommandsController()
                 _allCommands.Add(name, instance);
             }
         }
+
+        [Client]
+        public void ClientProcessCommand(string command)
+        {
+            CmdProcessCommand(command);
+        }
+
         /// <summary>
         /// Find and call command
         /// </summary>
         /// <param name="command">Command and it's args separated by spaces</param>
         /// <returns>Command response</returns>
-        public string ProcessCommand(string command)
+        [ServerRpc(RequireOwnership = false)]
+        public void CmdProcessCommand(string command, NetworkConnection conn = null)
         {
+            string ckey = Subsystems.Get<PlayerSystem>().GetCkey(conn);
+            if (!Subsystems.Get<PermissionSystem>().TryGetUserRole(ckey, out ServerRoleTypes userPermission))
+            {
+                RpcCommandAnswer(conn, string.Format("No role found for user {0}, can't process command", ckey));
+                return;
+            }
+
             string[] splitCommand = command.Split(' ');
-            string commandName = splitCommand[0]; 
+            string commandName = splitCommand[0];
             if (commandName == "help")
             {
-                return HelpCommand();
+                RpcCommandAnswer(conn, HelpCommand());
+                return;
             }
             if (splitCommand.Length > 1)
             {
                 if (splitCommand[1] == "help")
                 {
-                    return LongHelpCommand(command);
+                    RpcCommandAnswer(conn, LongHelpCommand(command));
+                    return;
                 }
             }
-            // Find proper command body and prepare correct args
-            if (_allCommands.ContainsKey(commandName))
+
+            if (!_allCommands.ContainsKey(commandName))
```

Comment:
```suggestion
            if (!_allCommands.TryGetValue(commandName, out var value))
```
This will allow us to get rid of access duplications like ```_allCommands[commandName]``` 
And change code below accordingly
```
if (value.AccessLevel > userPermission)
{
    RpcCommandAnswer(conn, "Access level too low, can't perform command");
    return;
}

if (value.ServerCommand)
{
    RpcCommandAnswer(conn, value.Perform(command.Split().Skip(1).ToArray(), conn));
    return;
}
```

The naming of a variable ```value``` might be not ideal and I would suggest to think more about it